ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/PullPhysicalPlan.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.physical.pull;
+
+import io.confluent.ksql.execution.streams.materialization.Locator.KsqlPartitionLocation;
+import io.confluent.ksql.execution.streams.materialization.Materialization;
+import io.confluent.ksql.internal.PullQueryExecutorMetrics;
+import io.confluent.ksql.physical.pull.operators.AbstractPhysicalOperator;
+import io.confluent.ksql.physical.pull.operators.DataSourceOperator;
+import io.confluent.ksql.query.QueryId;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.kafka.connect.data.Struct;
+
+public class PullPhysicalPlan {
```

Comment:
Javadoc for this class regarding 1) what metadata it contains, 2) what assumption it makes (e.g.  what `root` operator should be`) would be highly appreciated :)